src/Widgets/SpaceWidget/SpaceWidgetClass.js

```diff
@@ -0,0 +1,23 @@
+import * as Scrivito from "scrivito";
+
+const SpaceWidget = Scrivito.provideWidgetClass("SpaceWidget", {
+  attributes: {
+    size: [
+      "enum",
+      {
+        values: [
+          "free",
+          "xs-small",
+          "small",
+          "medium",
+          "large",
+          "xl-large",
+          "xxl-large",
+        ],
+      },
+    ],
+    freeSize: "string",
```

Comment:
I suggest do not support `freeSize` in the default Space-Widget. This also simplifies the code needed.